devicetypes/sky-nie/min-smart-plug.src/min-smart-plug.groovy

```diff
@@ -49,28 +68,62 @@ metadata {
         capability "Configuration"
         capability "Refresh"
         capability "Health Check"
+        capability "Button"
 
         attribute "firmwareVersion", "string"
         attribute "syncStatus", "string"
 
         fingerprint mfr: "0312", prod: "C000", model: "C009", deviceJoinName: "Minoston Outlet" // old MP21Z
         fingerprint mfr: "0312", prod: "FF00", model: "FF0C", deviceJoinName: "Minoston Outlet" //MP21Z Minoston Mini Smart Plug
         fingerprint mfr: "0312", prod: "AC01", model: "4001", deviceJoinName: "New One Outlet"  // N4001 New One  Mini Smart Plug
+        fingerprint mfr: "0312", prod: "EE00", model: "EE01", deviceJoinName: "Minoston Switch" //MS10ZS Minoston Smart Switch
+        fingerprint mfr: "0312", prod: "EE00", model: "EE03", deviceJoinName: "Minoston Switch" //MS12ZS Minoston Smart on/off Toggle Switch
+        fingerprint mfr: "0312", prod: "A000", model: "A005", deviceJoinName: "Evalogik Switch" //ZW30
+        fingerprint mfr: "0312", prod: "BB00", model: "BB01", deviceJoinName: "Evalogik Switch" //ZW30S Evalogik Smart on/off Switch
+        fingerprint mfr: "0312", prod: "BB00", model: "BB03", deviceJoinName: "Evalogik Switch" //ZW30TS Evalogik Smart on/off Toggle Switch
     }
 
     preferences {
-        configParams.each {
-            if (it.name) {
-                if (it.range) {
-                    input "configParam${it.num}", "number", title: "${it.name}:", required: false, defaultValue: "${it.value}", range: it.range
-                } else {
-                    input "configParam${it.num}", "enum", title: "${it.name}:", required: false, defaultValue: "${it.value}", options: it.options
-                }
-            }
-        }
+        getConfigParamInput(ledModeParam)
+        getConfigParamInput(autoOffIntervalParam)
+        getConfigParamInput(autoOnIntervalParam)
+        getConfigParamInput(powerFailureRecoveryParam)
+        input "disclaimer", "paragraph",
+                title: "WARNING",
+                description: "Configuring for 'Paddle Control'is only valid for the devices with product number of MS10ZS, MS12ZS, ZW30, ZW30S, ZW30TS(one of them)",
+                required: false
+        getConfigParamInput(paddleControlParam)
     }
 }
 
+private getConfigParamInput(param) {
+    if (param.range) {
+        input "configParam${param.num}", "number", title: "${param.name}:", required: false, defaultValue: "${param.value}", range: param.range
+    } else {
+        input "configParam${param.num}", "enum", title: "${param.name}:", required: false, defaultValue: "${param.value}", options: param.options
+    }
+}
+
+private addChildButton() {
+    log.warn "Creating Button Device"
+    def child = addChildDevice(
+            "sky-nie",
+            "Child Button",
```

Comment:
you could just as easily use the "smartthings" "Child Button" here and avoid having to create a separate DTH

Reply:
I checked the DTHs which call addChildDevice: in SmartThingsCommunity, the child devices are in the same namespace with their parents. But in some private repositories, there are also child devices those are not in the same namespace as the parent devices. Therefore, the current calling method is safe.

Reply:
I know, but we'd prefer to keep the code additions to a minimum.